src/api/apiMethods.js

```diff
@@ -55,54 +56,162 @@ const getApiMethodsSelector = () =>
+          return CellGroupType.SCROLLABLE;
+        }
+
+        return CellGroupType.FIXED_RIGHT;
+      };
+
+      const _getCellGroupTypeFromColumnGroupIndex = (columnGroupIndex) => {
+        const totalColumnGroups =
+          fixedColumnGroups.length +
+          scrollableColumnGroups.length +
+          fixedRightColumnGroups.length;
+
+        if (
+          columnGroupIndex < 0 ||
+          columnGroupIndex > totalColumnGroups ||
+          !Number.isInteger(columnGroupIndex)
+        ) {
+          throw `columnGroupIndex must be an integer between 0 and ${
+            totalColumnGroups - 1
+          } inclusive`;
+        }
+
+        if (columnGroupIndex < fixedColumnGroups.length) {
+          return CellGroupType.FIXED;
+        }
+
+        if (
+          columnGroupIndex <
+          fixedColumnGroups.length + scrollableColumnGroups.length
+        ) {
+          return CellGroupType.SCROLLABLE;
+        }
+
+        return CellGroupType.FIXED_RIGHT;
+      };
+
+      const _getColumnLocalIndex = (columnIndex, cellGroupType) => {
+        if (cellGroupType === CellGroupType.FIXED) {
+          return columnIndex;
+        }
+        if (cellGroupType === CellGroupType.SCROLLABLE) {
+          return columnIndex - fixedColumns.length;
+        } else if (cellGroupType === CellGroupType.FIXED_RIGHT) {
+          return columnIndex - fixedColumns.length - scrollableColumns.length;
+        }
+      };
+
+      const _getColumnGroupLocalIndex = (columnGroupIndex, cellGroupType) => {
+        if (cellGroupType === CellGroupType.FIXED) {
+          return columnGroupIndex;
+        }
+        if (cellGroupType === CellGroupType.SCROLLABLE) {
+          return columnGroupIndex - fixedColumnGroups.length;
+        } else if (cellGroupType === CellGroupType.FIXED_RIGHT) {
+          return (
+            columnGroupIndex -
+            fixedColumnGroups.length -
+            scrollableColumnGroups.length
+          );
+        }
+      };
+
       const getCellGroupWidth = (cellGroupType = CellGroupType.SCROLLABLE) => {
         const container = _getColumnContainerByCellGroupType(cellGroupType);
         return container.reduce((sum, column) => sum + column.width, 0);
       };
 
-      const getColumn = (
-        columnIndex,
-        cellGroupType = CellGroupType.SCROLLABLE
-      ) => {
+      const _getColumn = (localColumnIndex, cellGroupType) => {
         const container = _getColumnContainerByCellGroupType(cellGroupType);
-        return _getMinimalColumn(container[columnIndex]);
+        return _getMinimalColumn(container[localColumnIndex]);
+      };
+
+      const getColumn = (columnIndex) => {
+        const cellGroupType = _getCellGroupTypeFromColumnIndex(columnIndex);
+        const localColumnIndex = _getColumnLocalIndex(
+          columnIndex,
+          cellGroupType
+        );
+        return _getColumn(localColumnIndex, cellGroupType);
```

Comment:
Since the client is always gonna call these APIs in terms of the global column index, how about we only convert from global to local indices at the place where it's actually required?

i.e, `_.getColumn` should be the one who's doing the conversion since it directly uses the internal column lists that require local indices.

Reply:
Is this resolved?

Reply:
Sorry, i  missed that comment.   Addressed it now.